### DISCUSSION TITLE
Charge cycles in charge stats

### DISCUSSION THREAD

**Discussion by vkarthikbalaji** [teslamate-org/teslamate#418] [Ideas]
A charging cycle gives a good indication of the degradation performance of the battery. A charge cycle (per Apple) : Apple Inc. clarifies that a charge cycle means using all the battery's capacity, but not necessarily by discharging it from 100% to 0%: "You complete one charge cycle when you’ve used (discharged) an amount that equals 100% of your battery’s capacity — but not necessarily all from one charge. For instance, you might use 75% of your battery’s capacity one day, then recharge it fully overnight. If you use 25% the next day, you will have discharged a total of 100%, and the two days will add up to one charge cycle."[5]

In addition to the # charges / #  discharges, it would be informative to know how many full charge cycles the car has performed. 

This can be shown on the charge stats and eventually also on the projected range dashboard, as a toggle to visualise every 10 or 100 charge cycles

**Comment by marcogabriel** [comment]
The charge cycle value is nice, but it leads to false assumptions. Tesla states that a battery lasts 1500 charge cycles. In fact, this heavily depends on the min to max charge difference and also to the temperature while charging.

There are some papers who state, that if you go on a 75 kWh battery from
 
* 0% to 100% -> it may last 1500 cycles (=1500*75 kWh = 112500 kWh / ~200 Wh/km = ~562.500 km)
* 20% to 80% -> it lasts 2500 cycles (=2500*75 kWh = 187500 kWh / ~200 Wh/km = ~937.500 km)
* 40% to 60% -> it lasts 8000 cycles (=8000*75 kWh = 600000 kWh / ~200 Wh/km = ~3.000.000 km)

[Assumption: 200 Wh/km, 20°C. Unfortunately, I just can't find the source of the numbers that I stated above. But here's a good read: https://batteryuniversity.com/learn/article/how_to_prolong_lithium_based_batteries]

If taken this into consideration, a plain charge cycle value tells not much about the battery life. If charging statistics considered a factor and taken into account, you could even make a rough guess about the projected battery life.

**Comment by vkarthikbalaji** [comment]
Agree 100% @marcogabriel . Have read that battery university article quite a while ago. Indeed a good read. However, AFAIK, there is no universal truth on what is a charge cycle and also the behaviour of battery while charging/discharging is dependent on a lot of factors. Since we have the very nicely made charge heat map and the charge delta already, if we make the charge cycle as simple as possible (100% discharged irrespective of 100%-0% = 1cycle or 75%-25% + next day again 75%-25% = 1 cycle), then we would be having a baseline charge cycle for Teslamate. Assumptions (I would say comfortable assumptions) are always made in the world of science. So reading a charge cycle should always be within context of charge heatmap and charge delta.

I see having  a simple charge cycle atleast as a good starting point, given the variety of situations we encounter over the life of the car (battery ;)). I cannot come up with a better cycle definition if we have to take into account the charge deltas.
